lib/mongo/server/connection_pool.rb

```diff
@@ -1181,6 +1047,239 @@ def clear_pending_connections
           @pending_connections.delete(connection)
         end
       end
+
+      # The lock should be acquired when calling this method.
```

Comment:
👍

Reply:
do you think it would be a good idea to raise if the lock is not owned in some of these methods?

Reply:
Yes, it is a very good idea!

Reply:
done